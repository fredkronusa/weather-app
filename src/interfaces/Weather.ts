export interface IWeeklyForecastResponse {
  city_name: string;
  data: IForecastData[];
}

export interface IWeeklyForecast {
  weeklyForecast: IForecastData[];
}

export interface IWeather {
  description: string;
  icon: string;
}

export interface IForecastData {
  max_temp: number;
  min_temp: number;
  uv: number;
  valid_date: string;
  weather: IWeather;
}

export interface ICurrentForecastResponse {
  city_name: string;
  weather: IWeather;
  precip: number;
  rh: number;
  wind_spd: number;
  wind_cdir: string;
  ob_time: string;
  temp: number;
  uv: number;
}

export interface ICurrentForecast {
  city: string;
  weather: IWeather;
  precipitation: string;
  humidity: string;
  wind: string;
  date: string;
  temperature: number;
  uv: string;
}
