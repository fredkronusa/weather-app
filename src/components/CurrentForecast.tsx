import { colors, fonts } from "../common/styles";

import { ICurrentForecast } from "../interfaces/Weather";
//import { ICurrentForecast } from "../interfaces/weather";
import React from "react";
import { getTemperature } from "../utils/utils";
import styled from "styled-components";

const Location = styled.h2`
  margin-bottom: 0;
  padding-bottom: 0;
  font-size: ${fonts.xLarge};
  font-weight: normal;
`;

const LineItems = styled.span`
  color: ${colors.darkGrey};
  font-size: ${fonts.small};
  display: block;
`;

const Temperature = styled.div`
  margin-top: 40px;
  font-size: ${fonts.large};
  float: left;
  margin-right: 50px;
  /* TODO: handle this on a app level */
  @media only screen and (max-width: 3760px) {
    margin-left: 40px;
  }
`;

const Img = styled.img`
  float: left;
`;

const TempDisplay = styled.div`
  width: 100%;
  display: inline-block;
`;

const TempDisplayLeft = styled.div`
  width: 50%;
  float: left;
`;

const TempDisplayRight = styled(TempDisplayLeft)`
  margin-top: 20px;
`;

const InfoDisplay = styled.div`
  margin-left: 30px;
`;

interface ICurrentForecastComponent {
  currentForecast: ICurrentForecast;
  isCelsius: boolean;
}

export const CurrentForecastComponent: React.FC<ICurrentForecastComponent> = ({
  currentForecast,
  isCelsius,
}) => {
  return (
    <>
      <InfoDisplay>
        <Location>{currentForecast.city}</Location>
        <LineItems>{currentForecast.date}</LineItems>
        <LineItems>{currentForecast.weather.description}</LineItems>
      </InfoDisplay>
      <TempDisplay>
        <TempDisplayLeft>
          <Img
            src={`/images/icons/${currentForecast.weather.icon}.png`}
            alt={currentForecast.weather.description}
          />
          <Temperature
            aria-label={`current temperature in ${
              isCelsius ? "Celsius" : "Fahrenheit"
            } is `}
          >
            {getTemperature(currentForecast.temperature, isCelsius, true)}
            {isCelsius ? "\u2103" : "\u2109"}
          </Temperature>
        </TempDisplayLeft>
        <TempDisplayRight>
          <LineItems>Precipitation: {currentForecast.precipitation}</LineItems>
          <LineItems>Humidity: {currentForecast.humidity}</LineItems>
          <LineItems>Windy: {currentForecast.wind}</LineItems>
          <LineItems>Uv: {currentForecast.uv}</LineItems>
        </TempDisplayRight>
      </TempDisplay>
    </>
  );
};

export default CurrentForecastComponent;
