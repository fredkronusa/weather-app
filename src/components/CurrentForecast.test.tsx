import * as React from "react";

import CurrentForecast from "./CurrentForecast";
import { ICurrentForecast } from "../interfaces/Weather";
import { render } from "@testing-library/react";

const currentForecastMockData: ICurrentForecast = {
  city: "Melbourne",
  weather: {
    icon: "123",
    description: "foggy",
  },
  precipitation: "12%",
  humidity: "15%",
  wind: "my wind",
  date: "Tuesday, April 15th",
  temperature: 12,
  uv: "12",
};

describe("<CurrentForecast />", () => {
  // SNAPSHOT
  it("should match snapshot", async () => {
    const { asFragment } = render(
      <CurrentForecast
        currentForecast={currentForecastMockData}
        isCelsius={true}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
