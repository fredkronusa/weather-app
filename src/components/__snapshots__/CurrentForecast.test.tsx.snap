// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrentForecast /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-Axmtr jKxFqi"
  >
    <h2
      class="sc-AxjAm jGuQqP"
    >
      Melbourne
    </h2>
    <span
      class="sc-AxirZ jvZyRy"
    >
      Tuesday, April 15th
    </span>
    <span
      class="sc-AxirZ jvZyRy"
    >
      foggy
    </span>
  </div>
  <div
    class="sc-AxhUy iiarDM"
  >
    <div
      class="sc-AxgMl cgbCRu"
    >
      <img
        alt="foggy"
        class="sc-AxhCb wCzUk"
        src="/images/icons/123.png"
      />
      <div
        aria-label="current temperature in Celsius is "
        class="sc-AxiKw jFSTDB"
      >
        12nullâ„ƒ
      </div>
    </div>
    <div
      class="sc-AxgMl sc-AxheI bhRSaq"
    >
      <span
        class="sc-AxirZ jvZyRy"
      >
        Precipitation: 12%
      </span>
      <span
        class="sc-AxirZ jvZyRy"
      >
        Humidity: 15%
      </span>
      <span
        class="sc-AxirZ jvZyRy"
      >
        Windy: my wind
      </span>
      <span
        class="sc-AxirZ jvZyRy"
      >
        Uv: 12
      </span>
    </div>
  </div>
</DocumentFragment>
`;
